database/ddl.sql

```diff
@@ -0,0 +1,30 @@
+/* Copyright KubeArchive Authors
+SPDX-License-Identifier: Apache-2.0 */
+
+CREATE SCHEMA IF NOT EXISTS kubearchive;
+
+CREATE OR REPLACE FUNCTION kubearchive.trigger_set_timestamp()
+	RETURNS TRIGGER AS $$
+	BEGIN
+	  NEW.updated_at = NOW();
+	  RETURN NEW;
+	  END;
+	$$ LANGUAGE plpgsql;
+
+CREATE TABLE IF NOT EXISTS kubearchive.resource (
+    "uuid" uuid PRIMARY KEY,
+    "api_version" varchar NOT NULL,
+    "kind" varchar NOT NULL,
+    "name" varchar NOT NULL,
+    "namespace" varchar NOT NULL,
+    "resource_version" varchar NULL,
+    "created_at" timestamp NOT NULL DEFAULT now(),
+    "updated_at" timestamp NOT NULL DEFAULT now(),
+    "cluster_deleted_ts" timestamp NULL,
+    "data" jsonb NOT NULL
+    );
+
+CREATE OR REPLACE TRIGGER set_timestamp
+BEFORE UPDATE ON kubearchive.resource
```

Comment:
I thought we were going to set `created_at` via a trigger as well?

Reply:
We have `created_at` value automated, but as this should only be set when the registry is first created, the default value of the field gets the work done.

Note that we have both `created_at` and `updated_at` defaults to `now()`.

You can find more information in this PR https://github.com/kubearchive/kubearchive/pull/210 where I followed the steps from [this guide](https://medium.com/@alexishevia/automatic-timestamps-with-postgresql-8a5fc499d887)